core/src/main/java/software/amazon/awssdk/core/retry/conditions/RetryOnExceptionsCondition.java

```diff
@@ -42,40 +43,16 @@ private RetryOnExceptionsCondition(Set<Class<? extends Exception>> exceptionsToR
      */
     @Override
     public boolean shouldRetry(RetryPolicyContext context) {
-        if (context.exception() != null) {
-            for (Class exceptionClass : exceptionsToRetryOn) {
-                if (exceptionMatches(context, exceptionClass)) {
-                    return true;
-                }
-                // Note that we check the wrapped exception too because for things like SocketException or IOException
-                // we wrap them in an SdkClientException before throwing.
-                if (wrappedCauseMatches(context, exceptionClass)) {
-                    return true;
-                }
-            }
-        }
-        return false;
-    }
 
-    /**
-     * @param context        Context containing exception.
-     * @param exceptionClass Expected exception class.
-     * @return True if the exception in the context matches the provided class.
-     */
-    private boolean exceptionMatches(RetryPolicyContext context, Class exceptionClass) {
-        return context.exception().getClass().equals(exceptionClass);
-    }
-
-    /**
-     * @param context        Context containing exception.
-     * @param exceptionClass Expected exception class.
-     * @return True if the cause of the exception in the context matches the provided class.
-     */
-    private boolean wrappedCauseMatches(RetryPolicyContext context, Class exceptionClass) {
-        if (context.exception().getCause() == null) {
+        SdkException exception = context.exception();
+        if (exception == null) {
             return false;
         }
-        return context.exception().getCause().getClass().equals(exceptionClass);
+
+        //TODO: update equals to isAssignableFrom to match all sub classes of IOException
```

Comment:
Why not just do this now?

Reply:
We need to update `ResponseTransformer` to not retry SocketTimeoutException for `toOutputStream` for this change and write some more tests on this.

Keeping PR smaller to align with [`The Programmer's Oath`](http://blog.cleancoder.com/uncle-bob/2015/11/18/TheProgrammersOath.html) :-D

>I will make frequent, small, releases so that I do not impede the progress of others.